src/log/log.c

```diff
@@ -1972,11 +1976,11 @@ __wt_log_release(WT_SESSION_IMPL *session, WT_LOGSLOT *slot, bool *freep)
             __wt_verbose(session, WT_VERB_LOG,
               "log_release: sync log %s to LSN %" PRIu32 "/%" PRIu32, log->log_fh->name,
               sync_lsn.l.file, sync_lsn.l.offset);
-            WT_STAT_CONN_INCR(session, log_sync);
             time_start = __wt_clock(session);
             WT_ERR(__wt_fsync(session, log->log_fh, true));
             time_stop = __wt_clock(session);
             fsync_duration_usecs = WT_CLOCKDIFF_US(time_stop, time_start);
+            WT_STAT_CONN_INCR(session, log_sync);
```

Comment:
We want this change even if we throw away this debug branch. Actually, we should consider a `wt_log_release` refactor to call `wt_log_force_sync` to have one place where logging syncs.

Reply:
Raised WT-10206 to do the refactor. With your HAVE_DIAGNOSTIC change we are good to have this in.